[Do not merge] Test JupyterLab-LSP on RTC

hbcarlos opened the pull request:
As discussed in #590 this PR is for testing the extension on [JupyterLab RTC](https://github.com/jupyter-rtc/jupyterlab)

Modification on GitHub actions:

- Disabled linting (was linting jupyterlab's source code after cloning)
- Remove tests with nodejs < 12

jtpio commented on .github/workflows/job.test.yml:
```diff
@@ -54,17 +54,24 @@ jobs:
+          repository: jupyter-rtc/jupyterlab
+          path: ./jupyterlab
+      - name: Install JupyterLab RTC
+        run: pip install ./jupyterlab/
```
Is this enough do install from source, or do we also need a `jupyter lab build`?

krassowski replied:
Good question. The screenshots captured during the build suggest that we are running on the development version (with Isabela's CSS update), see below. Also, `python -m pip` might fix the smoke on MacOS (not that it is important).

![image](https://user-images.githubusercontent.com/5832902/117280939-09868580-ae5b-11eb-9e00-927dc9e39600.png)

krassowski replied:
We probably could script a test that opens Help → About and takes a screenshot.

jtpio replied:
We could also check the new `docprovider` and `shared-packages` are correctly included.

hbcarlos commented:
I changed the installation command to fix the smoke test and added the `--dev-mode` parameter to the `jupyter-lab` command (not sure if that's the right way to add a parameter, though)

jtpio commented on .github/workflows/job.test.yml:
```diff
@@ -135,27 +134,33 @@ jobs:
+          repository: jupyter-rtc/jupyterlab
+          path: ./jupyterlab
+      - name: Install JupyterLab RTC
+        run: python -m pip install ./jupyterlab/
```
Maybe we could now install the latest `3.1.0a7` prerelease?

hbcarlos replied:
Yep, I'll open a new PR for the prerelease and close this one.

krassowski commented:
@hbcarlos could you rebase now please? We fixed an issue with tests on master.

hbcarlos commented:
> @hbcarlos could you rebase now please? We fixed an issue with tests on master.

Sure, I'm on it.

krassowski commented:
As RTC is now merged and working well with LSP I will close this one. Thank you for testing this out.
